Ch04/drill4.cpp

```diff
@@ -0,0 +1,76 @@
+#include "../std_lib_facilities.h"
+
+double sum(const vector<double>& v) {
+	double sumofnum = 0.0;
+	for (double i:v) {
+		sumofnum += i;
+	};
+	return sumofnum;
+}
+
+bool supportedUnit(string unit) {
+	vector<string> units = {"m", "cm", "in", "ft"};
+
+	for (string i: units) {
+		if (unit == i) {
+			return true;
+		}
+	}
+	return false;
+}
+
+double conversionFactor(string unit) {
+	if (unit == "m") {
+		return 1;
+	} else if (unit == "cm") {
+		return 1.0/100;
+	} else if (unit == "in") {
+		return 2.54/100;
+	} else {
+		return 12*2.54/100;
+	}
+}
+
+int main () {
+	string sint1;
+	double int1;
+	vector<double> meters;
+
+	string unit;
+	
+	while (true) {
+		cout << "Type a measurement and a unit (m, cm, in, ft). ";
+		if (meters.size() > 0) {
+			cout << "\nYou may type '|' for the measurement to find out facts about your values and stop the program. ";
+		}
+		cin >> sint1 >> unit;
+		if (sint1 == "|") {
+			if (meters.size() > 0) {
+				break;
+			} else {
+				cout << "That is not a valid choice at this time.\n";
+				continue;
+			}
+		}
+		int1 = stod(sint1);
+
+		if (supportedUnit(unit)) {
+    		meters.push_back(int1 *= conversionFactor(unit));
+		} else {
+    		cout << "That is not a valid measurement.\n";
+    		continue;
+		}
+		sort(meters);
```

Comment:
For readability, I would move the `sort(meters)` inside the `if` statement above.